cve_bin_tool/version_scanner.py

```diff
@@ -40,7 +41,7 @@ def __init__(
         self.checkers = checkers or self.load_checkers()
         self.score = score
         self.total_scanned_files = 0
-        exclude_folders = [".git"]
+        self.exclude_folders = exclude_folders
         self.walker = DirWalk(folder_exclude_pattern=";".join(exclude_folders)).walk
```

Comment:
```suggest
        self.exclude_folders = exclude_folders
        self.exclude_folders.append(".git")
        self.walker = DirWalk(folder_exclude_pattern=";".join(self.exclude_folders)).walk
```

Reply:
I don't think user would ever want to scan `.git` directory. If we keep  [".git"] as default argument, we will only ignore it if user don't specify exclude_folder explicitly. So, if user wants to exclude folders then S/he has to write ".git" in list of exclude_folder explicitly.

Reply:
This isn't working because we are ignoring dirpath of file/directory. If you change it like following it will work
```python
self.pattern_match(os.path.join(dirpath, filename), os.path.normpath(self.folder_exclude_pattern))
```

Reply:
> This isn't working because we are ignoring dirpath of file/directory. If you change it like following it will work
> 
> ```python
> self.pattern_match(os.path.join(dirpath, filename), os.path.normpath(self.folder_exclude_pattern))
> ```

Oh yeah! Its working now. Thanks a lot!

Reply:
> ```python
> self.pattern_match(os.path.join(dirpath, filename), os.path.normpath(self.folder_exclude_pattern))
> ```

Tried some cases. It is still not working if the path of the directory and exclude path is the same. In that case, it shouldn't take any files into account. It works for all the files present in the subdirectory of exclude path though

Reply:
Push your latest code so that I can review it.